src/hooks/useVoucher/useVoucher.test.tsx

```diff
@@ -0,0 +1,256 @@
+      ]);
+    });
+
+    it("no update vouchers when invalid voucher is removed", async () => {
+      expect.assertions(2);
+      const { result } = renderHook(() => useVoucher(key, endpoint));
+
+      await wait(() => {
+        result.current.addVoucher({ serial: "123456789", denomination: 2 });
+        result.current.addVoucher({ serial: "123456788", denomination: 2 });
+      });
+
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 },
+        { serial: "123456788", denomination: 2 }
+      ]);
+
+      await wait(() => {
+        result.current.removeVoucher("invalid");
+      });
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 },
+        { serial: "123456788", denomination: 2 }
+      ]);
+    });
+
+    it("should maintain vouchers when vouchers are removed", async () => {
+      expect.assertions(3);
+      const { rerender, result } = renderHook(() => useVoucher(key, endpoint));
+
+      await wait(() => {
+        result.current.addVoucher({ serial: "123456789", denomination: 2 });
+        result.current.addVoucher({ serial: "123456788", denomination: 2 });
+      });
+
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 },
+        { serial: "123456788", denomination: 2 }
+      ]);
+
+      await wait(() => {
+        result.current.removeVoucher("123456788");
+      });
+
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 }
+      ]);
+      rerender([key, endpoint]);
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 }
+      ]);
+    });
+  });
+
+  describe("update vouchers when adding voucher", () => {
+    it("should update vouchers when new voucher are added", async () => {
+      expect.assertions(1);
+      const { result } = renderHook(() => useVoucher(key, endpoint));
+
+      await wait(() => {
+        result.current.addVoucher({ serial: "123456789", denomination: 2 });
+      });
+
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 }
+      ]);
+    });
+
+    it("should maintain vouchers when more vouchers are added", async () => {
+      expect.assertions(2);
+      const { rerender, result } = renderHook(() => useVoucher(key, endpoint));
+
+      await wait(() => {
+        result.current.addVoucher({ serial: "123456789", denomination: 2 });
+      });
+
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 }
+      ]);
+      rerender([key, endpoint]);
+      expect(result.current.vouchers).toStrictEqual([
+        { serial: "123456789", denomination: 2 }
+      ]);
+    });
+  });
+
+  describe("checkout cart", () => {
+    it("should set the correct checkoutResult when checkoutCart is called", async () => {
```

Comment:
will need to change this to the new name for the checkout, i suggest "checkoutVouchers"
```suggestion
    it("should set the correct checkoutResult when checkoutVouchers is called", async () => {
```